layouts.go

```diff
@@ -0,0 +1,101 @@
+package main
+
+import (
+	"fmt"
+	"image"
+	"strconv"
+	"strings"
+)
+
+// Layout contains the data to represent the layout of the widget.
+type Layout struct {
+	frames []image.Rectangle
+	size   int
+	margin int
+	height int
+}
+
+// NewLayout returns a new Layout with the accoriding size.
+func NewLayout(size int) *Layout {
+	margin := size / 18
+	height := size - (margin * 2)
+
+	return &Layout{
+		size:   size,
+		margin: margin,
+		height: height,
+	}
+}
+
+// DefaultLayout returns a layout that is evenly split in frameCount horziontal containers.
+func (l *Layout) DefaultLayout(frameCount int) []image.Rectangle {
+	if frameCount < 1 {
+		frameCount = 1
+	}
+	for i := 0; i < frameCount; i++ {
+		frame := l.defaultFrame(frameCount, i)
+		l.frames = append(l.frames, frame)
+	}
+	return l.frames
+}
+
+// FormatLayout returns a layout that is formatted according to frameReps.
+func (l *Layout) FormatLayout(frameReps []string, frameCount int) []image.Rectangle {
+	if frameCount < 1 {
+		frameCount = 1
+	}
+	for i := 0; i < frameCount; i++ {
+		if len(frameReps) < i+1 {
+			frame := l.defaultFrame(frameCount, i)
+			l.frames = append(l.frames, frame)
+			continue
+		}
+
+		frame, err := formatFrame(frameReps[i])
+		if err != nil {
+			fmt.Println("using default frame: ", err)
```

Comment:
Something we should comment out before merging.

Reply:
See my penultimate post

Reply:
Makes sense, thank you for clarifying!